velox/exec/tests/HashJoinTest.cpp

```diff
@@ -4669,6 +4670,87 @@ TEST_F(HashJoinTest, dynamicFiltersWithSkippedSplits) {
   }
 }
 
+TEST_F(HashJoinTest, dynamicFiltersAppliedToPreloadedSplits) {
+  vector_size_t size = 1000;
+  const int32_t numSplits = 5;
+  std::vector<RowVectorPtr> probeVectors;
+  probeVectors.reserve(numSplits);
+
+  std::vector<std::shared_ptr<TempFilePath>> tempFiles;
+  for (int32_t i = 0; i < numSplits; ++i) {
+    auto rowVector = makeRowVector({
+        makeFlatVector<int64_t>(
+            size, [&](auto row) { return (row + 1) * (i + 1); }),
+        makeFlatVector<int64_t>(size, [&](auto /*row*/) { return i; }),
+    });
+    probeVectors.push_back(rowVector);
+    tempFiles.push_back(TempFilePath::create());
+    writeToFile(tempFiles.back()->path, rowVector);
+  }
+  createDuckDbTable("u", probeVectors);
+
+  std::vector<RowVectorPtr> buildVectors{
+      makeRowVector({"c0"}, {makeFlatVector<int64_t>({0, numSplits})})};
+  createDuckDbTable("t", buildVectors);
+
+  auto makeInputSplits = [&](const core::PlanNodeId& nodeId) {
+    return [&] {
+      std::vector<exec::Split> probeSplits;
+      for (int32_t i = 0; i < numSplits; ++i) {
+        auto value = std::to_string(i);
+        auto split = facebook::velox::exec::test::HiveConnectorSplitBuilder(
+                         tempFiles[i]->path)
+                         .partitionKey("k", value)
+                         .build();
+        probeSplits.push_back(exec::Split(split));
+      }
+      SplitInput splits;
+      splits.emplace(nodeId, probeSplits);
+      return splits;
+    };
+  };
+  auto outputType =
+      ROW({"n1_0", "n1_1", "n1_2"}, {BIGINT(), BIGINT(), BIGINT()});
```

Comment:
These column names are hard to read. By convention, we use t0, t1, ... and u0, u1, ..... You can specify column names as first argument of makeRowVector.